server/routerlicious/packages/lambdas/src/nexus/index.ts

```diff
@@ -58,6 +59,14 @@ function getRoomId(room: IRoom) {
 	return `${room.tenantId}/${room.documentId}`;
 }
 
+function getClientRoomId(clientId: string) {
+	return `client#${clientId}`;
+}
+
+function isISentSignalMessage(message: any): message is ISentSignalMessage {
+	return typeof message === "object" && "targetClientId" in message;
```

Comment:
There are a few other important [properties in ISentSignalMessage](https://github.com/microsoft/FluidFramework/blob/main/common/lib/protocol-definitions/src/protocol.ts#L337-L382) that I think we should check for here.

Also, following the pattern for other type checks in the repo, `message` param should be `unknown` vs any.

Another thing to note is it could be better to use `typeof` checks because `"property" in JSON.parse(JSON.stringify({ property: undefined })) === false`, so for optional properties this could cause a v2 message like `{ content: "hello", clientConnectionNumber: 2 }` to be discarded as _not_ an `ISentSignalMessage`.

<img width="572" alt="image" src="https://github.com/microsoft/FluidFramework/assets/14362200/b146b10c-1eba-402e-a40b-93f43f0ef4c4">

This is a lot more verbose, and could maybe be simplified. One particular problem with `ISentSignalMessage` is that **all** of the properties are _optional_ or `unknown`, which means `message is ISentSignalMessage` says basically nothing.

## TL;DR

Perhaps something like this would be better for this particular use-case, since we aren't checking for any other possible properties in `ISentSignalMessage`?
```ts
function isTargetedSentSignalMessage(
  message: unknown,
): message is (ISentSignalMessage & Pick<Required<ISentSignalMessage>, "targetClientId">) {
  // is `message.content === undefined` valid?
  return typeof message === "object" &&
    typeof (message as ISentSignalMessage).targetClientId === "string";
}
```

Otherwise, we'd probably want to do something like this, where all of the optional properties are accounted for.

```ts
function isSentSignalMessage(
  message: unknown,
): message is ISentSignalMessage {
  return typeof message === "object" &&
    (typeof (message as ISentSignalMessage).content !== "undefined" ||
    typeof (message as ISentSignalMessage).targetClientId === "string" ||
    typeof (message as ISentSignalMessage).type === "string" ||
    typeof (message as ISentSignalMessage).clientConnectionNumber === "string" ||
    typeof (message as ISentSignalMessage).referenceSequenceNumber === "string");
}
```

Reply:
Looking at it now, I think the type check is unnecessar y and it should be sufficient to do the feature check to know it's an ISentSignalMessage -- that is what the feature check is for. [ODSP only does the feature check ](https://onedrive.visualstudio.com/OneDrive%20Service/_git/PushChannel?path=/src/server/roles/fluid/frontend/fluidWebSocketServer.ts&version=GBmaster&line=1218&lineEnd=1258&lineStartColumn=1&lineEndColumn=5&lineStyle=plain&_a=contents) so I think this pattern is good.

Reply:
So something like this: 
```typescript 
if (supportedFeaturesMap.get(clientId)?.[feature_submit_signals_v2]) {
    for (const signal of contentBatches as ISentSignalMessage[]) {
        const signalMessage: ISignalMessage = {
            clientId,
            content: signal.content,
            targetClientId: signal.targetClientId,
	};
    
        const roomId: string =
            signal.targetClientId !== undefined
	        ? getClientRoomId(signal.targetClientId)
	        : getRoomId(room);
    
	socket.emitToRoom(roomId, "signal", signalMessage);
    }
} else {
    contentBatches.forEach((contentBatch) => {
        const contents = Array.isArray(contentBatch)
		? contentBatch
		: [contentBatch];
	    for (const content of contents) {
		const signalMessage: ISignalMessage = {
			clientId,
			content,
		};
		const roomId: string = getRoomId(room);
    
		socket.emitToRoom(roomId, "signal", signalMessage);
	}
    });
}
```

Reply:
This makes much more sense to me. Thank you!